### PR TITLE
Fix several bugs in the run.py test harness that was causing freezing…

### PR DESCRIPTION
**Pull request by st-pasha**
… / task timeouts

---
This change is [Reviewable](https://reviewable.io/reviews/h2oai/h2o-3/504)

### CODE REVIEW

File: h2o-py/h2o/backend/connection.py [hunk 1]
```diff
@@ -290,9 +290,11 @@ def close(self):
         """
         if self._session_id:
             try:
+                # If the server gone bad, we don't want to wait forever...
+                self._timeout = 1
```
Comment (mmalohlava) [inline h2o-py/h2o/backend/connection.py]: Hardcoded timeout? Can we put it as an option into python config file?

**Comment by st-pasha** [comment]
@h2o-ops please test

File: scripts/run.py [hunk 2]
```diff
@@ -400,40 +400,29 @@ def start(self):
...
-                        f.close()
-                        print("H2O Cloud {} Node {} started with output file {}".format(self.cloud_num,
-                                                                                        self.node_num,
-                                                                                        self.output_file_name))
+        If successful, port number is stored in self.port; otherwise the
+        program is terminated. This call is blocking, and will wait for
+        up to 30s for the server to start up.
+        """
+        regex = re.compile(r"Open H2O Flow in your web browser: https?://([^:]+):(\d+)")
+        retries_left = 30
```
Comment (mmalohlava) [inline scripts/run.py]: again, it would be nice to control that when we have possibility to specify configuration file

File: scripts/run.py [hunk 3]
```diff
@@ -400,40 +400,29 @@ def start(self):
...
+        while retries_left and not self.terminated:
```
Comment (mmalohlava) [inline scripts/run.py]: Is this pattern useful somewhere else? as well? for example job progress? what if job progress `GET` call fails - do we try to retry?

File: scripts/run.py (hunk 2 above)
Comment (st-pasha) [inline scripts/run.py]: I agree, having a gazillion command line options is getting very ugly... And I wouldn't want to add another one. Having a config file would be much cleaner.

**Comment by st-pasha** [comment]
@h2o-ops please test!

**Comment by st-pasha** [comment]
@h2o-ops please test!

**Pull request merged**